### PR TITLE
add zookeeper name in system.replicas

### PR DESCRIPTION
**Pull request by ardenwick**
this would be useful for  queries like:
```sql
SELECT
    zookeeper_name,
    count()
FROM system.replicas
INNER JOIN system.parts USING (database, table)
WHERE database = 'business'
GROUP BY zookeeper_name
ORDER BY count() ASC
```

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add a new column `zookeeper_name` in system.replicas, to indicate on which (auxiliary) zookeeper cluster the replicated table's metadata is stored.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Ok. Need to add a test, see `tests/queries/0_stateless`.

**Comment by alexey-milovidov** [comment]
Need to add a test.

**Comment by cangyin** [comment]
sorry for being late, i 'm busy at work recently.

**Comment by alexey-milovidov** [comment]
Ok. Now we need a changelog entry "a user-readable short description of the changes that goes to CHANGELOG.md"

**Comment by cangyin** [comment]
> ### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
> add zookeeper name in system.replicas

there is the Change entry.

**Comment by alexey-milovidov** [comment]
But it is difficult to understand - we need to write what, and why.

**Comment by cangyin** [comment]
oh, I see.

**Comment by cangyin** [comment]
failing at CRLF again.

**Comment by cangyin** [comment]
some checks failed, seems irrelevant

File: tests/queries/0_stateless/02723_zookeeper_name.reference [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Create Tables
+Insert Data
+"t1","default",1
+"t2","default",2
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02723_zookeeper_name.reference:4]: The test contains a race condition: data parts can be merged before the test finishes. Therefore, the test will be flaky. Let's simplify it.

Comment (cangyin) [reply tests/queries/0_stateless/02723_zookeeper_name.reference]: fixed by removing the second `INSERT`

File: tests/queries/0_stateless/02723_zookeeper_name.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+CREATE TABLE t1(k UInt32, v UInt32) ENGINE ReplicatedMergeTree('/clickhouse/tables/{database}/test_02723/zookeeper_name/t1', '1') ORDER BY k;
+
+CREATE TABLE t2(k UInt32, v UInt32) ENGINE ReplicatedMergeTree('/clickhouse/tables/{database}/test_02723/zookeeper_name/t2', '1') ORDER BY k;
+
+SELECT 'Insert Data';
+
+INSERT INTO t1 SELECT * FROM generateRandom('k UInt32, v UInt32') LIMIT 1;
+INSERT INTO t2 SELECT * FROM generateRandom('k UInt32, v UInt32') LIMIT 1;
+
+SELECT
```
Comment (CheSema) [inline tests/queries/0_stateless/02723_zookeeper_name.sql:13]: That requst is flaky
Fail in master, for example.
https://s3.amazonaws.com/clickhouse-test-reports/35827/248e8819ff2c5fb9b78f32f8f0199f77c9a841b0/stateless_tests__debug__[5_5].html

Order by needed.

Comment (CheSema) [reply tests/queries/0_stateless/02723_zookeeper_name.sql:13]: https://github.com/ClickHouse/ClickHouse/pull/49592

**Pull request merged**